File: python/hash-table/longest-substring-without-repeating-characters.c
int lengthOfLongestSubstring(char* s) {
    int index[256] = {0};
    int last_index = -1;
    int len = strlen(s);
    int ret = 0;

    for(int i = 0; i < len; ++i){
        last_index = index[s[i]] > last_index ? index[s[i]] : last_index;
        index[s[i]] = i + 1;
        ret = (i - last_index + 1) > ret ? (i - last_index + 1) : ret;
    }

    return ret;
}

File: python/hash-table/longest-substring-without-repeating-characters.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        index = {}
        last_index = 0
        ret = 0
        for i in xrange(len(s)):
            if index.get(s[i],-1) >= last_index:
                last_index = index[s[i]] + 1
            index[s[i]] = i
            ret = max((i - last_index + 1),ret)

        return ret
